GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,619 @@
+# Governance documentation
+
+*Disclaimer: This is work in progress. Mechanisms are susceptible to change*
+
+This document describes the high-level architecture of the governance module. The governance module allows bonded Atom holders to vote on proposals on a 1 bonded Atom 1 vote basis.
+
+## Design overview
+
+The governance process is divided in a few steps that are outlined below:
+
+- **Proposal submission:** Proposal is submitted to the blockchain with a deposit 
+- **Vote:** Once deposit reaches a certain value (`MinDeposit`), proposal is confirmed and vote opens. Bonded Atom holders can then send `TxVote` transactions to vote on the proposal
+- If the proposal involves a software upgrade
+  - **Signal:** Validator start signaling that they are ready to switch to the new version
+  - **Switch:** Once more than 2/3rd validators have signaled their readiness to switch, their software automatically flips to the new version
+  
+## Proposal submission
+
+### Right to submit a proposal
+
+Any Atom holder, whether bonded or unbonded, can submit proposals by sending a `TxProposal` transaction. Once a proposal is submitted, it is identified by its unique `proposalID`.
+
+### Proposal filter (minimum deposit)
+
+To prevent spam, proposals must be submitted with a deposit in Atoms. Voting period will not start as long as the proposal's deposit is inferior to the minimum deposit parameter `MinDeposit`.
+
+When a proposal is submitted, it has to be acompagnied by a deposit that must be strictly positive but that can be inferior to `MinDeposit`. Indeed, the submitter need not pay for the entire deposit on its own. If a proposal's deposit is strictly inferior to `MinDeposit`, other Atom holders can increase the proposal's deposit by sending a `TxDeposit` transaction. Once the proposals's deposit reaches `minDeposit`, it enters voting period. 
+
+### Deposit refund
+
+There are two instances where Atom holders that deposited can claim back their deposit:
+- If the proposal is accepted
+- If the proposal's deposit does not reach `MinDeposit` for a period longer than `mMxDepositPeriod` (initial value: 2 months). Then the proposal is considered closed and nobody can deposit on it anymore.
```

Comment:
Do we really want to allow people to get their atoms back for a failed deposit? I think it might be simpler to just burn those. That would get rid of an extra parameter.

Reply:
It's also and especially for people who deposited on accepted proposals. The idea is to incentivize people to deposit on good proposals. I'm 100% in favour of keeping it at least if the proposal is accepted (other platforms such as Dfinity do it also). 

As for proposals that failed to reach `MinDeposit`, it's not that complicate once you have `TxClaimDeposit`. The idea is to not disincentivize people to deposit. They might get scared to deposit if they think there's a chance deposit will never reach MinDeposit. I think it's always good to advertise that there is a way to get your money back.

Reply:
I'm with adrian here, I'd like to burden the people who are pushing for proposals a bit... I'm okay if not that many proposals get put forth, less wasted time reading bad proposals